### PR TITLE
Added slide in component

### PR DESCRIPTION
**Pull request by Roene-JustBetter**

### CODE REVIEW

File: resources/views/components/slide-in/receipt.blade.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<x-rapidez-ct::slide-in>
+    <div class="absolute top-0 inset-x-0 h-1.5 bg-primary"></div>
```
Comment (jordythevulder) [inline resources/views/components/slide-in/receipt.blade.php:2]: this should be bg-ct-primary

Comment (Roene-JustBetter) [reply resources/views/components/slide-in/receipt.blade.php]: Fixed :)

**Review by jordythevulder — APPROVED** [review]

**Review by JimmyHoenderdaal — APPROVED** [review]

File: resources/js/components/SlideIn.vue [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        data() {
+            return {
+                target: null,
+                container: null,
+                left: '0',
+                top: '0',
+                isSwiping: false,
+                distanceX: 0,
+                distanceY: 0,
+                checked: false,
```
Comment (indykoning) [inline resources/js/components/SlideIn.vue:29]: Checked may be a bit too specific to our use case with a checkbox.
maybe something like `isOpen` is a better name here.

File: resources/js/components/SlideIn.vue [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                        }
+                        this.top = `${distance}px`
+                    } else {
+                        this.top = '0'
+                    }
+                }
+            },
+
+            onSwipeEnd() {
+                this.top = '0'
```
Comment (indykoning) [inline resources/js/components/SlideIn.vue:64]: Does setting this to `undefined` or `''` remove the style entirely? Maybe that is nicer so we can set the defaults in tailwind instead

File: resources/views/components/slide-in/index.blade.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<slide-in>
+    <template slot-scope="{ isSwiping, top, _self }">
+        <div ref="container">
+            <input v-if="_self" v-model="_self.checked" id="slide-in" type="checkbox" class="relative z-50 opacity-0 h-0 peer" />
+                <div
```
Comment (indykoning) [inline resources/views/components/slide-in/index.blade.php:5]: Indenting doesn't seem quite right here

**Review by indykoning — APPROVED** [review]

**Pull request merged**